### PR TITLE
feat(gitlab): add author date

### PR DESCRIPTION
**Pull request by warren830**
### Summary

Add authored date to mr_commit

### Does this close any open issues?

part of #4961 

### Screenshots

image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/bitbucket/models/migrationscripts/20230420_add_pr_commit_author_date.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	"github.com/apache/incubator-devlake/core/context"
+	"github.com/apache/incubator-devlake/core/errors"
+	"github.com/apache/incubator-devlake/core/plugin"
+	"github.com/apache/incubator-devlake/helpers/migrationhelper"
+)
+
+var _ plugin.MigrationScript = (*addGitlabCommitAuthoredDate)(nil)
+
+type GitlabMrCommit20230420 struct {
```
Comment (abeizn) [inline backend/plugins/bitbucket/models/migrationscripts/20230420_add_pr_commit_author_date.go:29]: bitbucket?

File: backend/plugins/gitlab/models/migrationscripts/20230420_add_mr_commit_author_date.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
```
Comment (abeizn) [inline backend/plugins/gitlab/models/migrationscripts/20230420_add_mr_commit_author_date.go:7]: lint?

File: backend/plugins/gitlab/models/mr_commit.go [hunk 3]
```diff
@@ -28,9 +29,10 @@ import (
 // Thus a "Merge Request Commit" needs to be considered as distinct from a "Commit"
 
 type GitlabMrCommit struct {
-	ConnectionId   uint64 `gorm:"primaryKey"`
-	CommitSha      string `gorm:"primaryKey;type:varchar(40)"`
-	MergeRequestId int    `gorm:"primaryKey;autoIncrement:false"`
+	ConnectionId       uint64 `gorm:"primaryKey"`
+	CommitSha          string `gorm:"primaryKey;type:varchar(40)"`
+	MergeRequestId     int    `gorm:"primaryKey;autoIncrement:false"`
+	CommitAuthoredDate *time.Time
```
Comment (abeizn) [inline backend/plugins/gitlab/models/mr_commit.go:35]: need *?

File: backend/plugins/gitlab/models/migrationscripts/20230420_add_mr_commit_author_date.go (hunk 2 above)
Comment (warren830) [reply backend/plugins/gitlab/models/migrationscripts/20230420_add_mr_commit_author_date.go:7]: fixed

File: backend/plugins/gitlab/models/mr_commit.go (hunk 3 above)
Comment (warren830) [reply backend/plugins/gitlab/models/mr_commit.go:35]: Either is ok, I think

**Review by abeizn — APPROVED** [review]
LGTM, bitbucket will update next pr

**Pull request merged**